src/main/java/frc/trigon/robot/subsystems/arm/simulationarm/SimulationArmIO.java

```diff
@@ -0,0 +1,98 @@
+package frc.trigon.robot.subsystems.arm.simulationarm;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.simulation.ElevatorSim;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import frc.trigon.robot.constants.RobotConstants;
+import frc.trigon.robot.subsystems.arm.ArmIO;
+import frc.trigon.robot.subsystems.arm.ArmInputsAutoLogged;
+import frc.trigon.robot.utilities.Conversions;
+
+public class SimulationArmIO extends ArmIO {
+    private final SingleJointedArmSim angleSimulation = SimulationArmConstants.ANGLE_MOTOR;
+    private final ElevatorSim elevatorSimulation = SimulationArmConstants.ELEVATOR_MOTOR;
+
+    private double angleMotorVoltage, elevatorMotorVoltage;
+
+    @Override
+    protected void updateInputs(ArmInputsAutoLogged inputs) {
+        angleSimulation.update(RobotConstants.PERIODIC_TIME_SECONDS);
+        elevatorSimulation.update(RobotConstants.PERIODIC_TIME_SECONDS);
+
+        inputs.angleMotorVoltage = angleMotorVoltage;
+        inputs.anglePositionDegrees = getAnglePositionDegrees();
+        inputs.angleVelocityDegreesPerSecond = getAngleVelocityDegreesPerSecond();
+        inputs.elevatorMotorVoltage = elevatorMotorVoltage;
+        inputs.elevatorPositionMeters = elevatorSimulation.getPositionMeters();
+        inputs.elevatorVelocityMetersPerSecond = elevatorSimulation.getVelocityMetersPerSecond();
+    }
+
+    @Override
+    protected void setTargetAngleState(TrapezoidProfile.State targetState) {
+        setAngleMotorsVoltage(calculateAngleVoltageFromState(targetState));
+    }
+
+    @Override
+    protected void setTargetElevatorState(TrapezoidProfile.State targetState) {
+        setElevatorMotorsVoltage(calculateElevatorVoltageFromState(targetState));
+    }
+
+    @Override
+    protected void stopAngleMotors() {
+        setAngleMotorsVoltage(0);
+    }
+
+    @Override
+    protected void stopElevatorMotors() {
+        setElevatorMotorsVoltage(0);
+    }
+
+    private double calculateElevatorVoltageFromState(TrapezoidProfile.State targetState) {
+        double pidOutput = SimulationArmConstants.ELEVATOR_PID_CONTROLLER.calculate(
+                elevatorSimulation.getPositionMeters(),
+                targetState.position
+        );
+        double feedforward = SimulationArmConstants.ELEVATOR_FEEDFORWARD.calculate(targetState.velocity);
+        return Conversions.compensatedPowerToVoltage(pidOutput + feedforward, SimulationArmConstants.VOLTAGE_COMPENSATION_SATURATION);
```

Comment:
pid and ff return an output. Based on how they're tuned, this could already return voltage based on the numbers, so no need for this.